diambra/arena/stable_baselines/make_sb_env.py

```diff
@@ -1,86 +1,94 @@
-        print("WARNING: running script without diambra CLI, this is a development option only.")
-        env_addresses = ["0.0.0.0:50051"]
+        raise Exception("ERROR: Running script without DIAMBRA CLI.")
 
     num_envs = len(env_addresses)
 
-    hardcore = False
-    if "hardcore" in env_settings:
-        hardcore = env_settings["hardcore"]
-
-    def _make_sb_env(rank):
-        def _thunk():
-            env = diambra.arena.make(env_settings["game_id"], env_settings,
-                                     wrappers_settings, traj_rec_settings,
-                                     seed=seed + rank, rank=rank)
-            if not hardcore:
+    # Seed management
+    if seed is None:
+        seed = int(time.time())
+    env_settings["seed"] = seed
 
-                # Applying custom wrappers
-                if custom_wrappers is not None:
-                    for wrap in custom_wrappers:
-                        env = wrap(env)
+    # Add the conversion from gymnasium to gym
+    old_gym_wrapper = [OldGymWrapper, {}]
+    if 'additional_wrappers_list' in wrappers_settings:
+        wrappers_settings['additional_wrappers_list'].insert(0, old_gym_wrapper)
+    else:
+        # If it's not present, add the key with a new list containing your custom element
+        wrappers_settings['additional_wrappers_list'] = [old_gym_wrapper]
 
-                env = AdditionalObsToChannel(env, key_to_add)
-            if p2_mode is not None:
-                if p2_mode == "integratedSelfPlay":
-                    env = IntegratedSelfPlay(env)
-                elif p2_mode == "selfPlayVsRL":
-                    env = SelfPlayVsRL(env, p2_policy)
-                elif p2_mode == "vsHum":
-                    env = VsHum(env, p2_policy)
+    def _make_sb_env(rank):
+        def _init():
+            env = diambra.arena.make(game_id, env_settings, wrappers_settings,
+                                     episode_recording_settings, render_mode, rank=rank)
 
             env = Monitor(env, logger.get_dir() and os.path.join(logger.get_dir(), str(rank)),
                           allow_early_resets=allow_early_resets)
             return env
-        return _thunk
-    set_global_seeds(seed)
+        set_global_seeds(seed)
+        return _init
 
     # If not wanting vectorized envs
     if no_vec and num_envs == 1:
-        return _make_sb_env(0)(), num_envs
+        env = _make_sb_env(0)()
+    else:
+        # When using one environment, no need to start subprocesses
+        if num_envs == 1 or not use_subprocess:
+            env = DummyVecEnv([_make_sb_env(i + start_index) for i in range(num_envs)])
+        else:
+            env = SubprocVecEnv([_make_sb_env(i + start_index) for i in range(num_envs)],
+                                start_method=start_method)
+
+    return env, num_envs
+
+class OldGymWrapper(gym.Wrapper):
+    def __init__(self, env):
+        """
+        Convert gymnasium to gym<=0.21 environment
```

Comment:
Why do we need that? Is that just for people who need to use Gym<=0.21? If so, why do we always seem to add this in make_env?

Reply:
Ah because sb <3 only supports gym <=0.21 anyway, right? Then you can resolve this comment

Reply:
Exactly! Only for the old SB. Thanks